blocks/rich-text/patterns.js

```diff
@@ -225,7 +225,11 @@ export default function( editor ) {
 		}
 
 		const block = pattern.transform( { content } );
+		onReplace( [ block ] );
 
-		editor.once( 'keyup', () => onReplace( [ block ] ) );
+		// WE call preventDefault and stopImmediatePropagation in order to prevent TinyMCE from creating additional newlines.
```

Comment:
Nit: Typo "WE" -> "We"

Reply:
Nit: `preventDefault` has nothing to do with TinyMCE. In fact, for the example from the original comment, the newline we're preventing is in a plain textarea rendered by the code block.

Reply:
Thank you for the feedback, it was applied.